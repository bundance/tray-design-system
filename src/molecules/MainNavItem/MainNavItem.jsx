import React from "react";

import HorizontalBox from "../../atoms/HorizontalBox/HorizontalBox";
import MainNavText from "../../atoms/MainNavText/MainNavText";
import Icon from "../../atoms/Icon/Icon";

import { faChevronRight } from "@fortawesome/free-solid-svg-icons";

const MainNavItem = ({ href, itemText, faIcon, onClick, selected }) => {
  const [state, setState] = React.useState("init");
  return (
    <HorizontalBox
      justifyContent="space-between"
      onMouseOver={() => setState("hover")}
      onMouseOut={() => setState("idle")}
      onClick={onClick}
      backgroundColor={state === "hover" ? "greyLight" : "inherit"}
      border={state === "hover" ? "1px solid greyLight" : "inherit"}
      borderRadius={4}
      paddingX={20}
    >
      <HorizontalBox paddingY={13} justifyContent="flex-start">
        <Icon
          faIcon={faIcon}
          height={22}
          color={
            selected || state === "hover"
              ? "mainNavSelectedText"
              : "textRegular"
          }
        />
        <MainNavText
          data-testid="main-nav-text"
          marginLeft={8}
          href={href}
          selected={selected}
          hover={state === "hover"}
        >
          {itemText}
        </MainNavText>
      </HorizontalBox>
      {(state === "hover" || selected) && (
        <Icon
          faIcon={faChevronRight}
          height={22}
          color={"mainNavSelectedText"}
        />
      )}
    </HorizontalBox>
  );
};
export default MainNavItem;
