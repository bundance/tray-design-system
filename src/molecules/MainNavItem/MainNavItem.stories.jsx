import React from "react";
import { storiesOf } from "@storybook/react";
import { faHome } from "@fortawesome/free-solid-svg-icons";

import MainNavItem from "./MainNavItem";

storiesOf("Molecules/MainNavItem", module)
  .add("Unselected", () => (
    <MainNavItem itemText="Dashboard" href="#" faIcon={faHome} />
  ))
  .add("Selected", () => (
    <MainNavItem itemText="Dashboard" href="#" faIcon={faHome} selected />
  ));
