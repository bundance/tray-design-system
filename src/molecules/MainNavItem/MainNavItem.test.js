import React from "react";
import { fireEvent, render } from "@testing-library/react";

import MainNavItem from "./MainNavItem";

import { faCoffee } from "@fortawesome/free-solid-svg-icons";

test("it renders correctly when not selected", () => {
  const { container } = render(
    <MainNavItem itemText="navItem" href="#" faIcon={faCoffee} />
  );
  expect(container.firstChild).toMatchSnapshot();
});

test("it renders correctly when selected", () => {
  const { container } = render(
    <MainNavItem itemText="navItem" href="#" faIcon={faCoffee} selected />
  );
  expect(container.firstChild).toMatchSnapshot();
});

test("it does not render a chevron when not selected", () => {
  const { getAllByTestId, debug } = render(
    <MainNavItem itemText="navItem" href="#" faIcon={faCoffee} />
  );

  const icons = getAllByTestId("icon");

  expect(icons.length).toBe(1);
});

test("it renders a chevron when selected", () => {
  const { getAllByTestId, debug } = render(
    <MainNavItem itemText="navItem" href="#" faIcon={faCoffee} selected />
  );

  const icons = getAllByTestId("icon");

  expect(icons.length).toBe(2);
});
