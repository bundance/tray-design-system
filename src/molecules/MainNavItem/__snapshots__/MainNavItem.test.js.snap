// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly when not selected 1`] = `
<div
  class="sc-AxjAm cvbzfR"
  display="flex"
>
  <div
    class="sc-AxjAm iuFNwh"
    display="flex"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-coffee fa-w-20 sc-AxhCb eWDCUd"
      color="textRegular"
      data-icon="coffee"
      data-prefix="fas"
      data-testid="icon"
      focusable="false"
      height="22"
      role="img"
      viewBox="0 0 640 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M192 384h192c53 0 96-43 96-96h32c70.6 0 128-57.4 128-128S582.6 32 512 32H120c-13.3 0-24 10.7-24 24v232c0 53 43 96 96 96zM512 96c35.3 0 64 28.7 64 64s-28.7 64-64 64h-32V96h32zm47.7 384H48.3c-47.6 0-61-64-36-64h583.3c25 0 11.8 64-35.9 64z"
        fill="currentColor"
      />
    </svg>
    <a
      class="sc-AxirZ fIyDrY"
      color="textRegular"
      data-testid="main-nav-text"
      font-family="Nunito Sans"
      font-size="7"
      font-weight="300"
      href="#"
      text-decoration="none"
    >
      navItem
    </a>
  </div>
</div>
`;

exports[`it renders correctly when selected 1`] = `
<div
  class="sc-AxjAm cvbzfR"
  display="flex"
>
  <div
    class="sc-AxjAm iuFNwh"
    display="flex"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-coffee fa-w-20 sc-AxhCb cjvlkc"
      color="mainNavSelectedText"
      data-icon="coffee"
      data-prefix="fas"
      data-testid="icon"
      focusable="false"
      height="22"
      role="img"
      viewBox="0 0 640 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M192 384h192c53 0 96-43 96-96h32c70.6 0 128-57.4 128-128S582.6 32 512 32H120c-13.3 0-24 10.7-24 24v232c0 53 43 96 96 96zM512 96c35.3 0 64 28.7 64 64s-28.7 64-64 64h-32V96h32zm47.7 384H48.3c-47.6 0-61-64-36-64h583.3c25 0 11.8 64-35.9 64z"
        fill="currentColor"
      />
    </svg>
    <a
      class="sc-AxirZ chDuZw"
      color="mainNavSelectedText"
      data-testid="main-nav-text"
      font-family="Nunito Sans"
      font-size="7"
      font-weight="600"
      href="#"
      text-decoration="none"
    >
      navItem
    </a>
  </div>
  <svg
    aria-hidden="true"
    class="svg-inline--fa fa-chevron-right fa-w-10 sc-AxhCb cjvlkc"
    color="mainNavSelectedText"
    data-icon="chevron-right"
    data-prefix="fas"
    data-testid="icon"
    focusable="false"
    height="22"
    role="img"
    viewBox="0 0 320 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
      fill="currentColor"
    />
  </svg>
</div>
`;
